### ISSUE TITLE
Tiles cached at wrong zoom level

### ISSUE THREAD

**Issue by juba** [bbox-services/bbox#87]
Hi,

We are using bbox to create a new version of the [lifemap](https://lifemap.univ-lyon1.fr/explore.html) website (version still in development, not public yet). It is working very smoothly, except for one issue : sometimes certain tiles are cached at the wrong zoom level, leading to display inconsistencies.

Here is an example screenshot: in this case the central tiles have been cached at a different (wrong) zoom level than the surrounding ones.

![image](https://github.com/user-attachments/assets/c8d0952f-aa0c-4e94-91d3-8cc6a792cc15)

When this problem arises, clearing the bbox tile cache solves the problem.

I'm sorry to not have a reproducible example to give, but maybe other people did encounter the same issue ? Or someone may have an idea about how to prevent this kind of problem ?

Many thanks in any case...

**Comment by pka** [comment]
You're using a file based cache, right? Are there any special settings like https://www.bbox.earth/docs/tile-server/reference/#cachelimit ?

**Comment by juba** [comment]
Yes, we use a file based cache, I don't think we have any special settings as our cache config is limited to:

```toml
[[tilestore]]
name = "tilecache"
[tilestore.files]
base_dir = "/opt/data/tilecache"
```

**Comment by FabianRechsteiner** [comment]
I can no longer find Gartenstrasse myself. Generally, you can zoom in anywhere, and you'll find tile errors if you pan around at a large scale.

![image](https://github.com/user-attachments/assets/e24ec9f8-bdc5-4992-b790-af590809ba75)

**Comment by juba** [comment]
I may have a live example on our development website, if it can be useful.
- go here: https://lifemap.cnrs.fr/tree?tool=search
- search for "archosauria"
- click on the corresponding search result

You should then arrive to a place where there is a tile discrepancy:

![image](https://github.com/user-attachments/assets/23af4707-543e-409f-9441-f6186aaf4965)

**Comment by pka** [comment]
This looks more like shifted by a few pixels, but with the correct zoom levels. When I zoom from the start page, I often see offsets between lines at tile borders. This looks like a numerical precision problem in tile generation?

**Comment by juba** [comment]
What I find strange is that these errors only appear for certain tiles at certain zoom levels, and they disappear (but may reappear elsewhere) if I clear the cache.

**Comment by pka** [comment]
By panning around in I found only tile errors like

![image](https://github.com/user-attachments/assets/d6b9aaad-173c-4f89-b743-41d7f86c1fef)

This looks also like the the zoom level is correct, but that some layers in the tiles are missing. Are there any database error messages?

**Comment by pka** [comment]
> What I find strange is that these errors only appear for certain tiles at certain zoom levels, and they disappear (but may reappear elsewhere) if I clear the cache.

Could you try once without cache?

**Comment by juba** [comment]
I've disabled server side caching on: 

https://lifemap.cnrs.fr/tree

I've tried navigating the map for a few minutes and for the moment I didn't encounter any discrepancies.

**Comment by FabianRechsteiner** [comment]
Caching has been disabled, and the old cache files have been deleted. However, errors still appear in the tiles. The issue does not seem to be related to caching but rather to the tile generation process. I haven’t found any errors in the logs.

![image](https://github.com/user-attachments/assets/b19bed67-25c9-4b01-9a83-52c29f16fd65)

**Comment by pka** [comment]
@juba I've no idea how caching can interfere here. Could you either send me the --loglevel debug output (with caching) or do a screen sharing session with me?

@FabianRechsteiner We should move your problems into an other issue. The last image looks like a `buffer_size` problem with thick lines?

**Comment by FabianRechsteiner** [comment]
I moved my `buffer_size` problem to the new issue https://github.com/bbox-services/bbox/issues/90#issue-2657906705

**Comment by juba** [comment]
Very sorry for my late answer.

I've tried some more experiments with different types of cache store and I get the following results:
- without cache, no discrepancies problem
- with a file cache, there are some discrepancies problems but they stay quite rare
- wit an mbtile cache, there are a lot of problems at almost every levels. This is the kind of rendering we get:

![2024-11-28 14 19 12 lifemap-back univ-lyon1 fr 4baa0e7baefc](https://github.com/user-attachments/assets/dacdfe20-d156-475d-abcc-8d64a5891af5)

Here is the tile server log corresponding to this rendering, I hope it can be useful.

[out.log](https://github.com/user-attachments/files/17948652/out.log)

And here is the corresponding `bbox.toml`.

[bbox.toml.txt](https://github.com/user-attachments/files/17948686/bbox.toml.txt)

Many thanks in advance for any insight...

**Comment by juba** [comment]
Sorry for my late answer...

I've tried to adjust some settings regarding to `buffer_size` in bbox and `RenderBuffer` in OpenLayers, and I haven't met any discrepancy for a while. So I can't be sure the issue has been solved, but at least I can't reproduce it for the moment.

**Comment by pka** [comment]
Please reopen, in case it can be reproduced again.
